crates/matrix-sdk-base/src/store/send_queue.rs

```diff
@@ -131,6 +132,9 @@ pub struct QueuedRequest {
     /// The bigger the value, the higher the priority at which this request
     /// should be handled.
     pub priority: usize,
+
+    /// The time that the request was original attempted.
+    pub created_at: Option<MilliSecondsSinceUnixEpoch>,
```

Comment:
Could this be non-optional, since this is not serialized we could always provide a value here? There might be places where we upgrade a `DependentQueuedRequest` with no `created_at` (because they were created before this change is introduced) to a `QueuedRequest`, but it seems fine to use `Milliseconds::now()` in those cases, maybe?

Reply:
Yeah, so if we are okay with injecting Milliseconds::now when it fails to parse coming out of the DB (either due to a previous migration or due to ... cosmic rays, then I think we can safely make this non optional.

IMO that's totally fine to do as you explained, just wasn't sure about injecting false data.

Reply:
I've updated it to use a ::now in most places and make it required everywhere except when it is exposed in the FFI (because SendHandle might not be present for the Local items)